### PR TITLE
Fix issues with CMake setup for http

### PR DESCRIPTION
**Pull request by aggarw13**
*Description of changes:*
* Add an `http` build target in CMake infrastructure
* Fix incorrect file-path to `generate_test_runner.rb` script in cmake script for CMock
* Fix `unity.h` include path dependency issue for `http_utest` target

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -1,63 +1,28 @@
...
+        )
+
+set_target_properties(cmock PROPERTIES
+        ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
+        POSITION_INDEPENDENT_CODE ON
+        COMPILE_FLAGS "-Og"
+        )
+
+add_library(unity STATIC
+        "${3RDPARTY_DIR}/CMock/vendor/unity/src/unity.c"
```
Comment (aggarg) [inline libraries/3rdparty/CMakeLists.txt:13]: Should we externalize unity and cmock paths in `unity_filepaths.cmake` and `cmock_filepaths.cmake` and include them, the way it is done for HTTP file paths.

File: libraries/standard/http/utest/CMakeLists.txt [hunk 2]
```diff
@@ -22,15 +22,20 @@ list(APPEND mock_define_list
 
 # ================= Create the library under test here (edit) ==================
 
+set(logging_sources
```
Comment (aggarg) [inline libraries/standard/http/utest/CMakeLists.txt:25]: Should these be in its own file?

File: libraries/standard/http/utest/CMakeLists.txt [hunk 3]
```diff
@@ -9,11 +9,11 @@ set(project_name "http")
 
 # list the files to mock here
 list(APPEND mock_list
-            ""
+            ${MODULES_DIR}/standard/http/third_party/http_parser/http_parser.h
```
Comment (aggarg) [inline libraries/standard/http/utest/CMakeLists.txt:12]: Why are parser files not part of `HTTP_SOURCES` in `filepaths.cmake`? Do we need a separate variable `HTTP_PARSER_SOURCES` and same for include paths in `filepaths.cmake`?

Comment (yourslab) [reply libraries/standard/http/utest/CMakeLists.txt:12]: I'm not so sure about adding `HTTP_PARSER_SOURCES` to `filepaths.cmake`. I was told by @leegeth to only include sources that are in both AFR and C SDK but share the same directory. AFR already has http_parser under `libraries/3rdparty`, so the directory would be different.

File: libraries/standard/http/utest/CMakeLists.txt (hunk 2 above)
Comment (yourslab) [reply libraries/standard/http/utest/CMakeLists.txt:25]: @aggarw13 We could add these to a separate file in `tools/cmake`.

File: libraries/3rdparty/CMakeLists.txt (hunk 1 above)
Comment (aggarw13) [reply libraries/3rdparty/CMakeLists.txt:13]: The utility of separating out the HTTP and MQTT file paths in their individual `filePaths.cmake` is their use in building the library targets (`mqtt`, `http`) and their tests targets (`mqtt_utest`, `http_utest`), both the categories that need lists of the sources and the header files.. It is not the same for Unity/CMock as we only need their library targets; we are not testing them.

File: libraries/standard/http/utest/CMakeLists.txt (hunk 2 above)
Comment (aggarw13) [reply libraries/standard/http/utest/CMakeLists.txt:25]: Yeah, will add them in the `tool/cmake` section

File: libraries/standard/http/utest/CMakeLists.txt (hunk 3 above)
Comment (aggarw13) [reply libraries/standard/http/utest/CMakeLists.txt:12]: For SSoT project, the `C-SDK` will be the SSoT for `HTTP`, thus, I don't think we need to be constrained by the directory structure of `HTTP` on the current version of AFR (as HTTP will be removed from it in the future, and only sourced from C-SDK)

Comment (leegeth) [reply libraries/standard/http/utest/CMakeLists.txt:12]: Source files and include directories for libraries(in this case http lib) will be used by build systems in repositories in which we use those libraries for SSoT. I have a question, how are we planning to build http_parser? Is it going to be added as a library and then a dependency to http lib or we include those as part of http library source itself?
If the option is the first one, for consistency, we can use filePaths.cmake for http_parser and use it for building http parser.

Comment (aggarw13) [reply libraries/standard/http/utest/CMakeLists.txt:12]: Ideally, just like `cmock` and `unity`, `http-parser` should have its own library target defined in its own directory

File: libraries/standard/http/utest/CMakeLists.txt (hunk 2 above)
Comment (aggarw13) [reply libraries/standard/http/utest/CMakeLists.txt]: Added

File: libraries/standard/http/utest/CMakeLists.txt (hunk 3 above)
Comment (aggarw13) [reply libraries/standard/http/utest/CMakeLists.txt:12]: Creation of a separate library for `http-parser` can be done in a separate PR. This PR was primarily meant for fixing issues in the build system blocking development

**Pull request merged**